src/WebAuthn.sol

```diff
@@ -122,47 +109,41 @@ library WebAuthn {
             return false;
         }
 
-        // 11. and 12. will be verified by the signature check
-        // 11. Verify that the value of C.type is the string webauthn.get.
+        string memory _type = webAuthnAuth.clientDataJSON.slice(webAuthnAuth.typeIndex, webAuthnAuth.typeIndex + 21);
+        if (keccak256(bytes(_type)) != EXPECTED_TYPE_HASH) {
+            return false;
+        }
+
         // 12. Verify that the value of C.challenge equals the base64url encoding of options.challenge.
-        string memory challengeB64url = Base64Url.encode(challenge);
-        string memory remainder = bytes(webAuthnAuth.crossOriginAndRemainder).length == 0
-            ? ""
-            : string.concat(",", webAuthnAuth.crossOriginAndRemainder);
-        string memory clientDataJSON = string.concat(
-            // A well formed clientDataJSON will always begin with
-            // {"type":"webauthn.get","challenge":"
-            // and so we can save calldata and use this by default
-            // https://www.w3.org/TR/webauthn/#clientdatajson-serialization
-            '{"type":"webauthn.get","challenge":"',
-            challengeB64url,
-            '",',
-            '"origin":"',
-            webAuthnAuth.origin,
-            '"',
-            remainder,
-            "}"
+        string memory challengeB64url = Base64.encode(challenge, true, true);
+        // 13. Verify that the value of C.challenge equals the base64url encoding of options.challenge.
+        bytes memory expectedChallenge = bytes(string.concat('"challenge":"', challengeB64url, '"'));
+        string memory actualChallenge = webAuthnAuth.clientDataJSON.slice(
+            webAuthnAuth.challengeIndex, webAuthnAuth.challengeIndex + expectedChallenge.length
         );
+        if (keccak256(bytes(actualChallenge)) != keccak256(expectedChallenge)) {
+            return false;
+        }
```

Comment:
One thing I found interesting is that the `slice()` function doesn't revert if the `start` or `end` indices are greater than the length of the string. Instead of reverting, it'll just return a shorter string, for example see this part of the Solady code:

```solidity
let subjectLength := mload(subject)
if iszero(gt(subjectLength, end)) { end := subjectLength }
if iszero(gt(subjectLength, start)) { start := subjectLength }
```

However I think this is fine. Because the only way that `hash(a) == hash(b)` is if `a` and `b` are equal. So if the user provides some weird values for `challengeIndex` or `typeIndex` such that it results in a shorter string, then the hashes aren't going to match, and verification will fail as expected. 

So I think this is all okay, just leaving this note